simulation/entities.py

```diff
@@ -60,8 +59,10 @@ def has_enough_conviction(self, funding_pool: float, token_supply: float, max_pr
 
 
 class Participant:
-    def __init__(self, holdings: TokenBatch):
-        self.sentiment = np.random.rand()
+    def __init__(self, holdings: TokenBatch, probability_func, random_number_func):
+        self._probability = probability_func
+        self._random_number = random_number_func
```

Comment:
same here, `self._random_number` sounds confusing. It is not yet a number! it is still a function.
For consistency, rename `self._probability` to `self._probability_func` as well.